core/src/main/kotlin/no/ntnu/beardblaster/screen/LoadingScreen.kt

```diff
@@ -1,78 +1,84 @@
-    }
-
-    override fun update(delta: Float) {
-        progress = Assets.assetManager.progress.roundToInt().toFloat()
-        // Check if assetManager is done loading
-        if (Assets.assetManager.update()) {
-            // Go to correct menu screen when done loading
-            if (UserAuth().isLoggedIn()) {
-                game.setScreen<MenuScreen>()
-            } else game.setScreen<LoginMenuScreen>()
+        KtxAsync.launch {
+            FontAsset.values().map { assets.loadAsync(it) }
+            AtlasAsset.values().map { assets.loadAsync(it) }
+            I18NAsset.values().map { assets.loadAsync(it) }
         }
+        font.data.scale(1.5f)
     }
 
     override fun render(delta: Float) {
-        update(delta)
-
-        // Draw the progress bar
-        shapeRenderer.use(ShapeRenderer.ShapeType.Filled, cam.combined)
-        {
-            it.rect(50f, (cam.viewportHeight / 2) - 12f, cam.viewportWidth - 50, 25f)
-            it.color = Color.RED
-            it.rect(50f, (cam.viewportHeight / 2) - 12f, progress * (cam.viewportWidth - 50), 25f)
+        val x = 200f
+        val height = 25f
+        val width = camera.viewportWidth - (x * 2)
+        val y = (camera.viewportHeight / 2) - (height / 2)
+        val progress = assets.progress.percent
+
+        renderer.use(ShapeRenderer.ShapeType.Filled, camera.combined) {
+            it.color = Color.PINK
+            it.rect(x, y, progress * width, height)
         }
 
-        // Draw the progress text
-        batch.use(cam.combined)
-        {
-            font.draw(it, (100 * progress).toString() + "% Loading Assets", 100f, 200f)
+        batch.use(camera.combined) {
+            font.draw(it, "Loading assets (${progress * 100}%)", x, y - 100f)
         }
-    }
 
-    override fun resize(width: Int, height: Int) {
-        viewport.update(width, height, true)
+        if (isFinishedLoading) {
+            Nls.i18nBundle = assets[I18NAsset.Default]
+            createSkin(assets)
+            addGameScreens()
+            when (UserAuth().isLoggedIn()) {
+                true -> game.setScreen<MenuScreen>()
+                false -> game.setScreen<LoginMenuScreen>()
+            }
+            game.removeScreen<LoadingScreen>()
+            dispose()
+        }
     }
 
-    fun queueAsset() {
-        Assets.loadTextures()
+    private fun addGameScreens() {
+        game.addScreen(LoginMenuScreen(game, batch, assets, camera))
+        game.addScreen(LoginScreen(game, batch, assets, camera))
+        game.addScreen(RegisterScreen(game, batch, assets, camera))
+        game.addScreen(MenuScreen(game, batch, assets, camera))
+        game.addScreen(TutorialScreen(game, batch, assets, camera))
+        game.addScreen(HighScoreScreen(game, batch, assets, camera))
+        game.addScreen(LobbyScreen(game, batch, assets, camera))
+        game.addScreen(JoinLobbyScreen(game, batch, assets, camera))
+        game.addScreen(GameplayScreen(game, batch, assets, camera))
```

Comment:
Can remove this from here and only add it in LobbyScreen when starting game

Reply:
That make sense

Reply:
On the other hand we need, for example,  access to the `HighScoreScreen` from the `MenuScreen`. I think I'll leave this as it is, and then we can refactor at a later stage.